### PR TITLE
Add 3rd party oauth example

### PR DESCRIPTION
**Pull request by horeaporutiu**
The example shows how to open a new browser tab, go through the OAuth process, return an authorization code, exchange that code for an access token, and then print that access token to the browser. We also then send a message to the front end to update the UI to show the user has logged in. 

You can watch the app demo w/ sound to see the app.

### CODE REVIEW

File: examples/3rd-party-oauth-login/README.md [hunk 1]
```diff
@@ -0,0 +1,103 @@
+# 3rd Party OAuth Login
+
+This app allows you to login to a 3rd party service using (you will need to provide OAuth URL) and tracks the logged in status via local storage.
```
Comment (mettin) [inline examples/3rd-party-oauth-login/README.md:3]:
```suggestion
This app allows you to login to a 3rd party service using (you will need to provide a OAuth URL) and track the logged in status via local storage.
```

File: examples/3rd-party-oauth-login/README.md [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+# 🏃🏽‍♂️ Run the app locally <a name="run"></a>
+
+1. Run `npm install` to install dependencies.
+2. Run `npm run start` to start the front end. We have a separate server for the backend. \
+   Your URL should be similar to this example:
+   ```
+   http://localhost:3000
+   ```
+3. Run `npm install` in the `src/backend` directory.
+4. Run `node app.js` in the `src/backend` directory.
```
Comment (mettin) [inline examples/3rd-party-oauth-login/README.md:48]:
```suggestion
   ```
3. `cd src/backend`
4. Run `npm install`.
5. Run `node app.js`
```

File: examples/3rd-party-oauth-login/README.md [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+5. If you need to use something like Ngrok for your redirectURL (I had to do this to go through the OAuth process for my Slack app) run `ngrok http 4000`.
```
Comment (mettin) [inline examples/3rd-party-oauth-login/README.md:49]: ngrok is written w/o a capital

File: examples/3rd-party-oauth-login/README.md [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+6. Your ngrok forwarding address should look something like: `https://bced-81-59-0-206.ngrok-free.app`. Then your <b>redirect URL</b> in the other service (for me it was in the App Settings in Slack) should be:
+   `https://bced-81-59-0-206.ngrok-free.app/redirect`
+7. Go into `src/backend` and fill in the `.sample.env` file with your OAuthURL, clientId, and clientSecret and
+   rename it to `.env` and then save the file.
```
Comment (mettin) [inline examples/3rd-party-oauth-login/README.md:53]:
```suggestion
7. Go into `src/backend` and fill in the `.sample.env` file with your OAuthURL, clientId, and clientSecret. Rename it to `.env` and then save the file.
```

File: examples/3rd-party-oauth-login/src/app.js [hunk 5]
```diff
@@ -0,0 +1,101 @@
+/* eslint-disable no-undef */
+// check eslint global config
+import "./assets/style.css";
+
+var loginBtn = document.getElementById("loginButton");
+var logoutBtn = document.getElementById("logoutButton");
+var loginText = document.getElementById("loginText");
```
Comment (mettin) [inline examples/3rd-party-oauth-login/src/app.js:7]:
```suggestion
const loginBtn = document.getElementById("loginButton");
const logoutBtn = document.getElementById("logoutButton");
const loginText = document.getElementById("loginText");
```

File: examples/3rd-party-oauth-login/src/app.js [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+// Define a function containing the code you want to run each time index.html is opened
+async function initialize() {
+  logoutBtn.style.display = "none";
+  await isLoggedIn();
+}
+
+// This function displays the user login status on the UI
+async function displayLoginStatus(loggedIn) {
+  try {
+    var statusParagraph = document.getElementById("loginStatus");
```
Comment (mettin) [inline examples/3rd-party-oauth-login/src/app.js:28]:
```suggestion
    const statusParagraph = document.getElementById("loginStatus");
```

File: examples/3rd-party-oauth-login/src/app.js [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+    console.log(error);
+  }
+}
+
+// This function checks the local storage for the user's id. Local storage is just for demo purposes.
+// It is recommended to implement your own storage for a production application.
+async function isLoggedIn() {
+  try {
+    // Check if the user is logged in by checking the local storage on the browser
+    const loggedIn = localStorage.isLoggedIn;
```
Comment (mettin) [inline examples/3rd-party-oauth-login/src/app.js:54]: does this work? Should it be `localStorage.getItem('isLoggedIn')`?

File: examples/3rd-party-oauth-login/src/app.js [hunk 8]
```diff
@@ -0,0 +1,101 @@
...
+    return false; // Handle errors gracefully
+  }
+}
+
+//this function opens a new browser tab to start the OAuth process
+//we then add an event listener to the window to listen for the message from the backend for once the OAuth flow is completed
+async function startOAuthFlow() {
+  try {
+    const response = await fetch("http://localhost:4000");
+    let OAuthURL = await response.json();
```
Comment (mettin) [inline examples/3rd-party-oauth-login/src/app.js:70]:
```suggestion
    const OAuthURL = await response.json();
```

**Review by mettin — COMMENTED** [review]
Small comment: Since you're using slack in the example i'd mention that in the title too: `Login using a 3rd party app - Slack`

File: examples/3rd-party-oauth-login/README.md (hunk 1 above)
Comment (horeaporutiu) [reply examples/3rd-party-oauth-login/README.md:3]: changed to the following:

This app allows you to login to a 3rd party service using OAuth (you will need to provide an OAuth URL) and tracks the logged in status via local storage.

File: examples/3rd-party-oauth-login/README.md (hunk 3 above)
Comment (horeaporutiu) [reply examples/3rd-party-oauth-login/README.md:49]: fixed

File: examples/3rd-party-oauth-login/README.md (hunk 2 above)
Comment (horeaporutiu) [reply examples/3rd-party-oauth-login/README.md:48]: fixed

File: examples/3rd-party-oauth-login/src/app.js (hunk 6 above)
Comment (horeaporutiu) [reply examples/3rd-party-oauth-login/src/app.js:28]: fixed

File: examples/3rd-party-oauth-login/src/app.js (hunk 8 above)
Comment (horeaporutiu) [reply examples/3rd-party-oauth-login/src/app.js:70]: fixed

File: examples/3rd-party-oauth-login/src/app.js (hunk 7 above)
Comment (horeaporutiu) [reply examples/3rd-party-oauth-login/src/app.js:54]: changed it to

    const loggedIn = localStorage.getItem("isLoggedIn") === "true";

and this seems to work fine.

File: examples/3rd-party-oauth-login/APP_SUBMISSION.md [hunk 9]
```diff
@@ -0,0 +1,5 @@
+## Submission to Miro Marketplace
+
+Congrats! You have finished building your app & you'd like to publish it for
+users. You can submit your app on the
```
[the remark this answers is not in the document]
Comment (mettin) [reply examples/3rd-party-oauth-login/APP_SUBMISSION.md]: If thats the case, can you create a ticket to update CMA?

File: examples/3rd-party-oauth-login/README.md (hunk 1 above)
Comment (mettin) [reply examples/3rd-party-oauth-login/README.md]: Still wrong I think. 
This app allows you to ...
AND tracks

Maybe it should be 'and IT tracks'?

File: examples/3rd-party-oauth-login/APP_SUBMISSION.md (hunk 9 above)
[the remark this answers is not in the document]
Comment (horeaporutiu) [reply examples/3rd-party-oauth-login/APP_SUBMISSION.md]: Here you go boss: https://miro.atlassian.net/browse/DX-2915 @mettin

File: examples/3rd-party-oauth-login/README.md (hunk 1 above)
Comment (horeaporutiu) [reply examples/3rd-party-oauth-login/README.md]: the IT is implied :) 

It is grammatically correct as is now, but I can change it to "IT tracks".

Comment (horeaporutiu) [reply examples/3rd-party-oauth-login/README.md]: @mettin I split it into two sentences for easier comprehension. 


This app allows you to login to a 3rd party service using OAuth (you will need to provide an OAuth URL). The app tracks if the user
has logged in or not via [localStorage](https://developer.mozilla.org/en-US/docs/Web/API/Window/localStorage).

File: examples/3rd-party-oauth-login/README.md (hunk 4 above)
Comment (horeaporutiu) [reply examples/3rd-party-oauth-login/README.md]: fixed

**Review by mettin — APPROVED** [review]
seal-of-gif

**Pull request merged**